boost 1.75.0 incompatibilty

I recently updated to `boost 1.75.0` and encountered the following issue during compilation of `deal.II`:
```
In file included from /usr/include/boost/geometry/core/ring_type.hpp:27,
                 from /usr/include/boost/geometry/core/point_type.hpp:24,
                 from /usr/include/boost/geometry/core/coordinate_system.hpp:22,
                 from /usr/include/boost/geometry/core/cs.hpp:26,
                 from /home/marc/dealii/include/deal.II/base/point.h:26,
                 from /home/marc/dealii/include/deal.II/base/bounding_box.h:23,
                 from /home/marc/dealii/include/deal.II/boost_adaptors/bounding_box.h:21,
                 from /home/marc/dealii/source/fe/mapping.cc:17:
/usr/include/boost/geometry/strategies/envelope/services.hpp: In instantiation of ‘struct boost::geometry::strategies::envelope::services::default_strategy<boost::geometry::model::multi_point<dealii::Point<1, double>, std::vector, std::allocator>, dealii::BoundingBox<1, double>, boost::geometry::cartesian_tag>’:
/usr/include/boost/geometry/algorithms/detail/envelope/interface.hpp:94:21:   required from ‘static void boost::geometry::resolve_strategy::envelope<boost::geometry::default_strategy, false>::apply(const Geometry&, Box&, boost::geometry::default_strategy) [with Geometry = boost::geometry::model::multi_point<dealii::Point<1, double>, std::vector, std::allocator>; Box = dealii::BoundingBox<1, double>]’
/usr/include/boost/geometry/algorithms/detail/envelope/interface.hpp:116:52:   required from ‘static void boost::geometry::resolve_variant::envelope<Geometry>::apply(const Geometry&, Box&, const Strategy&) [with Box = dealii::BoundingBox<1, double>; Strategy = boost::geometry::default_strategy; Geometry = boost::geometry::model::multi_point<dealii::Point<1, double>, std::vector, std::allocator>]’
/usr/include/boost/geometry/algorithms/detail/envelope/interface.hpp:197:47:   required from ‘void boost::geometry::envelope(const Geometry&, Box&) [with Geometry = boost::geometry::model::multi_point<dealii::Point<1, double>, std::vector, std::allocator>; Box = dealii::BoundingBox<1, double>]’
/home/marc/dealii/include/deal.II/base/bounding_box.h:407:28:   required from ‘dealii::BoundingBox<spacedim, Number>::BoundingBox(const Container&) [with Container = boost::container::small_vector<dealii::Point<1, double>, 2, void, void>; int spacedim = 1; Number = double]’
/home/marc/dealii/source/fe/mapping.cc:76:12:   required from ‘dealii::BoundingBox<spacedim> dealii::Mapping<dim, spacedim>::get_bounding_box(const typename dealii::Triangulation<dim, spacedim>::cell_iterator&) const [with int dim = 1; int spacedim = 1; typename dealii::Triangulation<dim, spacedim>::cell_iterator = dealii::TriaIterator<dealii::CellAccessor<1, 1> >]’
/home/marc/dealii/build/source/fe/mapping.inst:11:17:   required from here
/usr/include/boost/geometry/strategies/envelope/services.hpp:32:5: error: static assertion failed: Not implemented for this coordinate system.
   32 |     BOOST_GEOMETRY_STATIC_ASSERT_FALSE(
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This is triggered in an explicit instantiation of `Mapping` with codim = 1
```
template class Mapping< 1 ,  2 >;
```

Can you confirm this?

---

I tried to follow the trace of this failed assertion. We grab vertices from a cell here ...
https://github.com/dealii/dealii/blob/691ff4907964605dd21e94f06c880786af2cd612/source/fe/mapping.cc#L26-L39

.. and create a corresponding `BoundingBox` around them here ...
https://github.com/dealii/dealii/blob/691ff4907964605dd21e94f06c880786af2cd612/source/fe/mapping.cc#L68-L77

... with the following wrapper function.
https://github.com/dealii/dealii/blob/691ff4907964605dd21e94f06c880786af2cd612/include/deal.II/base/bounding_box.h#L403-L411

The remaining part lies in the `boost` library.

---

I am not particularly sure what causes this assertion to fail. Could it be that the `codim=1` case is now being treated differenly in `boost`?

--- a/include/deal.II/base/bounding_box.h
+++ b/include/deal.II/base/bounding_box.h
@@ -26,6 +26,9 @@
 DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
 #include <boost/geometry/algorithms/envelope.hpp>
 #include <boost/geometry/geometries/multi_point.hpp>
+#if DEAL_II_BOOST_VERSION_GTE(1, 75, 0)
+#  include <boost/geometry/strategies/envelope/cartesian.hpp>
+#endif
 DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
 
 DEAL_II_NAMESPACE_OPEN

--- a/include/deal.II/base/config.h.in
+++ b/include/deal.II/base/config.h.in
@@ -238,6 +238,13 @@
 #define DEAL_II_BOOST_VERSION_MINOR @Boost_MINOR_VERSION@
 #define DEAL_II_BOOST_VERSION_SUBMINOR @Boost_SUBMINOR_VERSION@
 
+#define DEAL_II_BOOST_VERSION_GTE(major,minor,subminor) \
+ ((DEAL_II_BOOST_VERSION_MAJOR * 100000 + \
+    DEAL_II_BOOST_VERSION_MINOR * 100 + \
+     DEAL_II_BOOST_VERSION_SUBMINOR) \
+    >= \
+    (major)*100000 + (minor)*100 + (subminor))
+
 /*
  * Gmsh:
  */

